### ISSUE TITLE
somthing about training samples

### ISSUE THREAD

**Issue by depth-perception** [AlexeyAB/darknet#2610]
thanks for your great works! recently,i use the darknet to detect the defect in a picture，i only use the defect samples but there are also some zero defect samples which I don't use. what should I do to make full use of the zero  defect samples to improve the map of the detector?

**Comment by AlexeyAB** [comment]
> but there are also some zero defect samples which I don't use. 

Do you mean that you have training images without objects?

Just add paths to these images to the `train.txt` file, and create **empty** `<image-filename>.txt` file for each image, as described here: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> desirable that your training dataset include images with non-labeled objects that you do not want to detect - negative samples without bounded box (empty .txt files) - use as many images of negative samples as there are images with objects

**Comment by depth-perception** [comment]
@AlexeyAB i  have 946 training images , I augments the images to 3340，there are also 2250 images，i put all of 2250 images as the negative samples to the training dataset，but the map improves a very little，why？
how many negative images should i use camparing 3340 images？

**Comment by AlexeyAB** [comment]
> how many negative images should i use camparing 3340 images？

50% / 50%

So it is enought to use 3340 negative samples.

**Comment by shartoo** [comment]
i did add negative samples without bounded box to my `train.txt` but the loss curse appears strange.
![avg-loss](https://user-images.githubusercontent.com/5716282/67823859-a8605200-faff-11e9-8f19-fe66e926da68.png)
and the training output as below
![training_output](https://user-images.githubusercontent.com/5716282/67823890-c1690300-faff-11e9-8dc3-cb834bf2b57e.png)
what's wrong operation has i done?

**Comment by shartoo** [comment]
And the validation result in `./results` for five class are all 0kb which means the model detected nothing when using `*_last.weights`,but there are detection result when usinig earlier model.

**Comment by AlexeyAB** [comment]
What cfg-file do you use?
Look at the files bad.list and bad_label.list if they exist.

**Comment by shartoo** [comment]
`yolov3.cfg` is my cfg file,i fixed this problem by change my batch_size from 4 to 16 now.Thank you!

```
[net]
# Testing
batch=4
subdivisions=4
# Training
batch=16
subdivisions=8
width=608
height=608
```

**Comment by AlexeyAB** [comment]
Use
```
[net]
batch=64
subdivisions=32
width=608
height=608
```

instead of
```
[net]
# Testing
batch=4
subdivisions=4
# Training
batch=16
subdivisions=8
width=608
height=608

```


as described here: https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects

**Comment by shartoo** [comment]
Thank you ,it works.
